File: bins/public_bin/dl-ubuntuforum-thread.pl
#!/usr/bin/env perl

# Script originally from unutbu on http://ubuntuforums.org/showpost.php?p=4782850&postcount=880
# requires lynx

# Usage:
# dl-ubuntuforum-thread.pl "http://ubuntuforums.org/showthread.php?t=442945" 
use strict;
use warnings;

my $thread=shift @ARGV;
warn "thread=$thread\n";
my $page=1;
my $current_page;
my $max_page=2;
my $cmd;
my ($thread_num)=($thread=~m/\?t=(\d+)/);
warn "thread_num=$thread_num\n";
while (1) {
    last if ($page>$max_page);
    $cmd="wget -O thread${thread_num}-${page}.html \"http://ubuntuforums.org/printthread.php?t=$thread_num&pp=75&page=${page}\"";

    print "$cmd\n";
    system($cmd);
    open(FILE,"<","thread${thread_num}-${page}.html") || die; 
    while (<FILE>) {
	if (/Page (\d+) of (\d+)/) {
	    ($current_page,$max_page)=($1,$2);
	    print "Page $current_page of $max_page\n";
	    last;
	}
    }
    close(FILE);
    $cmd="lynx -dump thread${thread_num}-${page}.html > thread${thread_num}-${page}.txt";
    print "$cmd\n";
    system($cmd);
    $page++;
}

$cmd="cat " . join(' ', map {"thread${thread_num}-${_}.txt"} (1..$max_page)) . "> thread${thread_num}.txt";
print "$cmd\n";
system($cmd);

for $page (1..$max_page) {
    for my $ext (qw(txt html)) {
	my $file="thread${thread_num}-${page}.${ext}";
	$cmd="unlink($file)";
	print "$cmd\n";
	unlink($file);
    }    
}

$cmd="bzip2 thread${thread_num}.txt";
system($cmd);

File: bins/public_bin/dl-ubuntuforum-thread.py
#!/usr/bin/env python

# Script originally from TironN on http://ubuntuforums.org/showpost.php?p=9313875&postcount=1879
# requires lynx

import urllib2
import sys
import os
import re

site = sys.argv[1]
page = 1
max_page = 25
site_num = re.findall('(?<=\?t\=)[\d]+', site)[0]
onerun = 0

print "Thread number = %s" % site_num
while page <= max_page:
    
    cmd = 'wget -O thread-%s-%s.html \'http://ubuntuforums.org/printthread.php?t=%s&pp=75&page=%s\'' % (site_num, page, site_num, page)
        
    print "\nNow running: " + cmd
    os.system(cmd)
        
    if onerun == 0:
        file = open('thread-%s-%s.html' % (site_num, page))
        found = re.search('Page\ (\d+)\ of\ (\d+)', file.read())
        page = int(found.group(1))
        max_page = int(found.group(2))
        print max_page
        print page
        onerun = 1
    
        file.close()
    
    cmd = 'lynx -dump thread-%s-%s.html > thread-%s-%s.txt' % (site_num, page, site_num, page)
    print "\nNow running: " + cmd
    os.system(cmd)
        
    page += 1



file_list = []
for i in range(1, (max_page + 1)):
    
    file_list.append('thread-%s-%s.txt ' % (site_num, i))
    
    print file_list

    cmd = 'cat '
for i in file_list:
    cmd = cmd + i + " "

cmd = cmd + '> thread-%s.txt' % site_num

print "Now running: " + cmd
os.system(cmd)

ext = ['html', 'txt']

for p in range(1, (max_page + 1)):
    for e in ext:
        file = 'thread-%s-%d.%s' % (site_num, p, e)
        print file
        os.remove(file)
        
cmd = 'bzip2 thread-%s.txt' % site_num
os.system(cmd)
